### PR TITLE
Limit the length of parameterization testcase name to 255 characters.

### PR DESCRIPTION
**Pull request by yuxuan-ms**
## Bug / Requirement Description
Clearly and concisely describe the problem.

## Solution description
Describe your code changes in detail for reviewers.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: tests/functional/testplan/testing/multitest/test_parametrization.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import sys
 import logging
-from contextlib import contextmanager
 from unittest import mock
-from imp import reload
+from importlib import reload
```
Comment (zhenyu-ms) [inline tests/functional/testplan/testing/multitest/test_parametrization.py:4]: 👍

File: testplan/testing/multitest/suite.py [hunk 2]
```diff
@@ -580,10 +580,10 @@ def _validate_testcase(func):
         raise exc
 
     if len(func.name) > defaults.MAX_TEST_NAME_LENGTH:
-        warnings.warn(
-            'Name defined for testcase "{}" is too long,'
-            ' consider customizing testcase name with argument "name_func"'
-            " in @testcase decorator.".format(func.__name__)
+        raise ValueError(
+            f'Name defined for testcase "{func.name}" is longer than 255,'
```
Comment (Pyifan) [inline testplan/testing/multitest/suite.py:584]: use const

File: testplan/testing/multitest/parametrization.py [hunk 3]
```diff
@@ -267,7 +271,7 @@ def _check_tag_func(tag_func):
         )
 
 
-def _parametrization_name_func_wrapper(func_name, kwargs):
+def _parametrization_name_func_wrapper(func_name: str, kwargs: dict):
     """
```
Comment (Pyifan) [inline testplan/testing/multitest/parametrization.py:275]: update the doc string to tell the diff of these 2 functions

File: testplan/testing/multitest/parametrization.py [hunk 4]
```diff
@@ -291,6 +295,31 @@ def _parametrization_name_func_wrapper(func_name, kwargs):
...
+                f"it is of type: {type(generated_name)}, value: {generated_name}"
+            )
+        if len(generated_name) <= MAX_METHOD_NAME_LENGTH:
+            return generated_name
+        else:
+            warnings.warn(
+                f"The name name_func returned ({generated_name}) is too long, using index suffixed names."
+            )
+
+    generated_name = f"{name} {index}"
```
Comment (Pyifan) [inline testplan/testing/multitest/parametrization.py:319]: is the name here function name or user specified @test_case (name=...) param

File: doc/newsfragments/3160_changed.case_name.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Limit the length of parameterization testcase name to 255 characters. If the name length exceeds 255 characters, index-suffixed names (e.g., {func_name[:240]} 1, {func_name[:240]} 2) will be used.
```
Comment (Pyifan) [inline doc/newsfragments/3160_changed.case_name.rst:1]: no truncation

**Review by zhenyu-ms — APPROVED** [review]
LGTM

**Pull request merged**